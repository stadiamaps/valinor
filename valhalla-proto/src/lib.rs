mod proto;

pub use proto::*;

// Re-export protobuf lib (for the traits)
pub use prost;

pub fn add(left: u64, right: u64) -> u64 {
    left + right
}

#[cfg(test)]
mod tests {
    use super::*;
    use prost::Message;

    #[test]
    fn decode_api_message() {
        const MESSAGE: &[u8] = &[
            0xa, 0x43, 0x12, 0x5, 0x65, 0x6e, 0x2d, 0x55, 0x53, 0x40, 0xc, 0x50, 0x0, 0x58, 0x0,
            0x6a, 0xf, 0x8, 0x0, 0x12, 0xb, 0xa, 0x3, 0xb8, 0x3, 0x0, 0x1a, 0x4, 0x6e, 0x6f, 0x6e,
            0x65, 0xb8, 0x1, 0x0, 0xc5, 0x1, 0x0, 0x0, 0x80, 0x3f, 0xd0, 0x1, 0x0, 0xa8, 0x2, 0x0,
            0xb8, 0x2, 0x0, 0xc8, 0x2, 0x0, 0xe0, 0x2, 0x1, 0x80, 0x3, 0x0, 0x90, 0x3, 0x0, 0xb0,
            0x3, 0xff, 0xff, 0xff, 0xff, 0xf, 0x22, 0xa3, 0x1, 0x32, 0x5, 0x33, 0x2e, 0x35, 0x2e,
            0x31, 0x38, 0xee, 0xc5, 0xc4, 0xc5, 0x6, 0x42, 0x6, 0x73, 0x74, 0x61, 0x74, 0x75, 0x73,
            0x42, 0x8, 0x63, 0x65, 0x6e, 0x74, 0x72, 0x6f, 0x69, 0x64, 0x42, 0x9, 0x65, 0x78, 0x70,
            0x61, 0x6e, 0x73, 0x69, 0x6f, 0x6e, 0x42, 0x11, 0x74, 0x72, 0x61, 0x6e, 0x73, 0x69,
            0x74, 0x5f, 0x61, 0x76, 0x61, 0x69, 0x6c, 0x61, 0x62, 0x6c, 0x65, 0x42, 0x10, 0x74,
            0x72, 0x61, 0x63, 0x65, 0x5f, 0x61, 0x74, 0x74, 0x72, 0x69, 0x62, 0x75, 0x74, 0x65,
            0x73, 0x42, 0xb, 0x74, 0x72, 0x61, 0x63, 0x65, 0x5f, 0x72, 0x6f, 0x75, 0x74, 0x65,
            0x42, 0x9, 0x69, 0x73, 0x6f, 0x63, 0x68, 0x72, 0x6f, 0x6e, 0x65, 0x42, 0xf, 0x6f, 0x70,
            0x74, 0x69, 0x6d, 0x69, 0x7a, 0x65, 0x64, 0x5f, 0x72, 0x6f, 0x75, 0x74, 0x65, 0x42,
            0x12, 0x73, 0x6f, 0x75, 0x72, 0x63, 0x65, 0x73, 0x5f, 0x74, 0x6f, 0x5f, 0x74, 0x61,
            0x72, 0x67, 0x65, 0x74, 0x73, 0x42, 0x6, 0x68, 0x65, 0x69, 0x67, 0x68, 0x74, 0x42, 0x5,
            0x72, 0x6f, 0x75, 0x74, 0x65, 0x42, 0x6, 0x6c, 0x6f, 0x63, 0x61, 0x74, 0x65,
        ];
        let result = Api::decode(MESSAGE).expect("Failed to decode message");

        if !cfg!(miri) {
            insta::assert_debug_snapshot!(result);
        }
    }
}
