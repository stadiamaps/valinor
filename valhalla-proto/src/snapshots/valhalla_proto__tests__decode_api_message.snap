---
source: valhalla-proto/src/lib.rs
expression: result
---
Api {
    options: Some(
        Options {
            units: Kilometers,
            directions_type: Instructions,
            format: Json,
            action: Status,
            costing_type: None,
            costings: {
                0: Costing {
                    r#type: None,
                    has_options: Some(
                        Options(
                            Options {
                                filter_stop_action: NoAction,
                                filter_stop_ids: [],
                                filter_operator_action: NoAction,
                                filter_operator_ids: [],
                                filter_route_action: NoAction,
                                filter_route_ids: [],
                                exclude_edges: [],
                                fixed_speed: 0,
                                axle_count: 0,
                                use_lit: 0.0,
                                disable_hierarchy_pruning: false,
                                ignore_non_vehicular_restrictions: false,
                                use_truck_route: 0.0,
                                exclude_bridges: false,
                                exclude_tunnels: false,
                                exclude_tolls: false,
                                exclude_highways: false,
                                exclude_ferries: false,
                                has_maneuver_penalty: None,
                                has_destination_only_penalty: None,
                                has_gate_cost: None,
                                has_gate_penalty: None,
                                has_toll_booth_cost: None,
                                has_toll_booth_penalty: None,
                                has_alley_penalty: None,
                                has_country_crossing_cost: None,
                                has_country_crossing_penalty: None,
                                has_ferry_cost: None,
                                has_avoid_bad_surfaces: None,
                                has_use_ferry: None,
                                has_use_highways: None,
                                has_use_tolls: None,
                                has_use_roads: None,
                                has_max_distance: None,
                                has_walking_speed: None,
                                has_step_penalty: None,
                                has_max_grade: None,
                                has_max_hiking_difficulty: None,
                                has_mode_factor: None,
                                has_walkway_factor: None,
                                has_sidewalk_factor: None,
                                has_alley_factor: None,
                                has_driveway_factor: None,
                                has_driveway_penalty: None,
                                has_transit_start_end_max_distance: None,
                                has_transit_transfer_max_distance: None,
                                has_transport_type: None,
                                has_top_speed: None,
                                has_use_hills: None,
                                has_use_primary: None,
                                has_use_trails: None,
                                has_low_class_penalty: None,
                                has_hazmat: None,
                                has_weight: None,
                                has_axle_load: None,
                                has_height: None,
                                has_width: None,
                                has_length: None,
                                has_cycling_speed: None,
                                has_wheelchair: None,
                                has_bicycle: None,
                                has_use_bus: None,
                                has_use_rail: None,
                                has_use_transfers: None,
                                has_transfer_cost: None,
                                has_transfer_penalty: None,
                                has_flow_mask: Some(
                                    FlowMask(
                                        0,
                                    ),
                                ),
                                has_bike_share_cost: None,
                                has_bike_share_penalty: None,
                                has_rail_ferry_cost: None,
                                has_use_rail_ferry: None,
                                has_ignore_restrictions: None,
                                has_ignore_oneways: None,
                                has_ignore_access: None,
                                has_ignore_closures: None,
                                has_shortest: None,
                                has_service_penalty: None,
                                has_use_tracks: None,
                                has_use_distance: None,
                                has_use_living_streets: None,
                                has_service_factor: None,
                                has_closure_factor: None,
                                has_private_access_penalty: None,
                                has_exclude_unpaved: None,
                                has_include_hot: None,
                                has_include_hov2: None,
                                has_include_hov3: None,
                                has_exclude_cash_only_tolls: None,
                                has_restriction_probability: None,
                                has_elevator_penalty: None,
                                has_hgv_no_access_penalty: None,
                                has_max_allowed_speed_limit: None,
                            },
                        ),
                    ),
                    has_name: Some(
                        Name(
                            "none",
                        ),
                    ),
                    has_filter_closures: None,
                },
            },
            locations: [],
            exclude_locations: [],
            sources: [],
            targets: [],
            date_time_type: NoTime,
            shape: [],
            contours: [],
            trace: [],
            shape_match: WalkOrSnap,
            filter_action: NoAction,
            filter_attributes: [],
            recostings: [],
            exclude_polygons: [],
            expansion_properties: [],
            pbf_field_selector: None,
            reverse: false,
            banner_instructions: false,
            elevation_interval: 0.0,
            voice_instructions: false,
            dedupe: false,
            admin_crossings: false,
            has_language: Some(
                Language(
                    "en-US",
                ),
            ),
            has_id: None,
            has_jsonp: None,
            has_encoded_polyline: None,
            has_range: Some(
                Range(
                    false,
                ),
            ),
            has_verbose: Some(
                Verbose(
                    false,
                ),
            ),
            has_date_time: None,
            has_resample_distance: None,
            has_polygons: Some(
                Polygons(
                    false,
                ),
            ),
            has_denoise: Some(
                Denoise(
                    1.0,
                ),
            ),
            has_generalize: None,
            has_show_locations: Some(
                ShowLocations(
                    false,
                ),
            ),
            has_gps_accuracy: None,
            has_search_radius: None,
            has_turn_penalty_factor: None,
            has_breakage_distance: None,
            has_use_timestamps: Some(
                UseTimestamps(
                    false,
                ),
            ),
            has_shape_format: None,
            has_alternates: Some(
                Alternates(
                    0,
                ),
            ),
            has_interpolation_distance: None,
            has_guidance_views: Some(
                GuidanceViews(
                    false,
                ),
            ),
            has_height_precision: None,
            has_roundabout_exits: Some(
                RoundaboutExits(
                    true,
                ),
            ),
            has_linear_references: None,
            has_prioritize_bidirectional: Some(
                PrioritizeBidirectional(
                    false,
                ),
            ),
            has_expansion_action: None,
            has_skip_opposites: Some(
                SkipOpposites(
                    false,
                ),
            ),
            has_matrix_locations: Some(
                MatrixLocations(
                    4294967295,
                ),
            ),
        },
    ),
    trip: None,
    directions: None,
    status: Some(
        Status {
            version: "3.5.1",
            tileset_last_modified: 1756439278,
            available_actions: [
                "status",
                "centroid",
                "expansion",
                "transit_available",
                "trace_attributes",
                "trace_route",
                "isochrone",
                "optimized_route",
                "sources_to_targets",
                "height",
                "route",
                "locate",
            ],
            has_has_tiles: None,
            has_has_admins: None,
            has_has_timezones: None,
            has_has_live_traffic: None,
            has_bbox: None,
            has_has_transit_tiles: None,
            has_osm_changeset: None,
        },
    ),
    matrix: None,
    isochrone: None,
    expansion: None,
    info: None,
}
